conda/install.py

```diff
@@ -504,47 +514,40 @@ def delete_trash(prefix):
         trash_dir = join(pkg_dir, '.trash')
         try:
             log.debug("Trying to delete the trash dir %s" % trash_dir)
-            rm_rf(trash_dir, max_retries=1)
+            rm_rf(trash_dir, max_retries=1, trash=False)
         except OSError as e:
             log.debug("Could not delete the trash dir %s (%s)" % (trash_dir, e))
 
-def move_to_trash(prefix, f, tempdir=None):
+def move_to_trash(path):
     """
-    Move a file f from prefix to the trash
-
-    tempdir should be the name of the directory in the trash
+    Move a path to the trash
     """
+    # Try deleting the trash every time we use it.
+    delete_trash()
+
     from conda import config
 
     for pkg_dir in config.pkgs_dirs:
+        import tempfile
         trash_dir = join(pkg_dir, '.trash')
 
-        try:
+        if not isdir(trash_dir):
             os.makedirs(trash_dir)
-        except OSError as e1:
```

Comment:
Oops, I missed the fact that you just ignored errors when it wasn't a 'file already exists'